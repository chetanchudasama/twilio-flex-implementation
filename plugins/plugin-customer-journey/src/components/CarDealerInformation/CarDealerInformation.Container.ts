import { connect } from "react-redux";
import { Dispatch } from "redux";
import { getWorkerIsAvailableForOutboundCall } from "../../states/selectors";
import { AppState } from "../../states";
import { DispatchToProps, StateToProps } from "./CarDealerInformation.Props";
import CarDealerInformation from "./CarDealerInformation";

const mapStateToProps = (state: AppState): StateToProps => {
  return {
    canMakeCall: getWorkerIsAvailableForOutboundCall(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CarDealerInformation);
