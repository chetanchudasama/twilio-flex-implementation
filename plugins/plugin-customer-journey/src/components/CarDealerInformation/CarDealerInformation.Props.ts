export interface StateToProps {
  canMakeCall: boolean;
}

export interface DispatchToProps {}
