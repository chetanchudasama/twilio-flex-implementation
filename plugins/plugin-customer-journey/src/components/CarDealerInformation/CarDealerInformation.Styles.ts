import { secondary, black } from "@common/components";
import styled from "react-emotion";

export const CarDealerInformationStyles = styled("div")({
  marginTop: 8,
  ".dealer-name": {
    fontSize: "16px",
    fontWeight: 600,
    paddingBottom: "8px",
    marginLeft: 5,
  },
  ".info-icon": {
    color: secondary,
    opacity: 0.4,
  },
  ".dealer-info": {
    fontSize: "14px",
    padding: "2px 10px 3px 10px",
    wordBreak: "break-word",
  },
  ".dealer-info-url": {
    fontSize: "14px",
    wordBreak: "break-word",
    padding: "2px 0px 3px 10px",
  },
  ".dealer-info-url a": {
    color: black,
  },
  ".dealer-phone, dealer-address": {
    display: "-webkit-box",
    WebkitBoxOrient: "vertical",
    WebkitLineClamp: 2,
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
});
