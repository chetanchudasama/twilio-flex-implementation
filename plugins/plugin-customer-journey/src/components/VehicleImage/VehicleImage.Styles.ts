import {
  white,
  borderStyle,
  defaultSnackbarColor,
  darkGrey,
} from "@common/components";
import styled from "react-emotion";

export const VehicleImageStyles = styled("div")({
  width: "100%",
  height: "100%",
  border: borderStyle,
  position: "relative",
  backgroundColor: defaultSnackbarColor,
  borderRadius: "10px 0px 0px 3px",

  ".vehicle-img": {
    width: "100%",
    height: "100%",
    maxWidth: "100%",
    maxHeight: "100%",
    objectFit: "contain",
    cursor: "pointer",
  },
  ".image-count": {
    position: "absolute",
    left: "0",
    top: "0",
    display: "flex",
    justifyContent: "center",
    padding: "10px",
    color: white,
    background: darkGrey,
    cursor: "pointer",
    borderRadius: "10px 0px 5px 0px",
  },
  ".icon": {
    fontSize: "18px",
  },
  ".image-count .icon": {
    paddingRight: "8px",
  },
  ".image-missing": {
    width: "100%",
    height: "100%",
  },
});
