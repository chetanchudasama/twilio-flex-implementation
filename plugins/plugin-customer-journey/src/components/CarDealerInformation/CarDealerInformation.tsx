import { Grid } from "@material-ui/core";
import LanguageIcon from "@material-ui/icons/Language";
import LocationOnIcon from "@material-ui/icons/LocationOn";
import StarIcon from "@material-ui/icons/Star";
import { Actions, IconButton } from "@twilio/flex-ui";
import React from "react";
import { DealerResponseModel } from "../../models/DealerResponseModel";
import { StateToProps, DispatchToProps } from "./CarDealerInformation.Props";
import { CarDealerInformationStyles } from "./CarDealerInformation.Styles";

export interface CarDealerInformationProp {
  dealerInfo: DealerResponseModel;
  distance?: number;
}

export type Props = StateToProps & DispatchToProps & CarDealerInformationProp;

const CarDealerInformation: React.FC<Props> = ({
  dealerInfo,
  distance,
  canMakeCall,
}) => {
  const triggerOutboundCall = () => {
    if (canMakeCall && dealerInfo.phone) {
      const payload = {
        destination: dealerInfo.phone,
      };
      Actions.invokeAction("StartOutboundCall", payload);
    }
  };

  return (
    <CarDealerInformationStyles>
      <Grid container>
        <Grid item xs={12} className="dealer-name">
          {dealerInfo.dealerName}
        </Grid>
        <Grid item container>
          <Grid item xs={4}>
            <Grid container>
              <Grid item xs={1}>
                <LocationOnIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={11} className="dealer-info dealer-address">
                {dealerInfo.town + (distance ? ` (${distance} miles)` : "")}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={2}>
            <Grid container>
              <Grid item xs={2}>
                <StarIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={10} className="dealer-info dealer-rating">
                {dealerInfo.rating ?? " -"}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={4}>
            <Grid container>
              <Grid item xs={1}>
                <IconButton
                  icon="Call"
                  className="info-icon"
                  disabled={!canMakeCall}
                  onClick={() => triggerOutboundCall()}
                />
              </Grid>
              <Grid item xs={11} className="dealer-info dealer-phone">
                {dealerInfo.phone}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={2}>
            <Grid container>
              <Grid item xs={2}>
                <LanguageIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={10} className="dealer-info-url">
                <a href={dealerInfo.url} target="blank">
                  Website
                </a>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </CarDealerInformationStyles>
  );
};

export default CarDealerInformation;
