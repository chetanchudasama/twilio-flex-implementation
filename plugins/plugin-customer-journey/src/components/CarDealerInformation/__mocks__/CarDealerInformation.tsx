import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
