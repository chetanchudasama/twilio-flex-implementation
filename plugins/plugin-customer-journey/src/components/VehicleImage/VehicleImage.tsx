import { ImageCarousel, CarNotFoundIcon } from "@common/components";
import React, { useState, useEffect } from "react";
import CameraAltIcon from "@material-ui/icons/CameraAlt";
import { VehicleImageStyles } from "./VehicleImage.Styles";

export interface VehicleImageProps {
  image: string;
  imageList?: string[];
  imagesCount?: number;
  showImageCount?: boolean;
}

export const VehicleImage: React.FC<VehicleImageProps> = ({
  image,
  imageList,
  imagesCount,
  showImageCount,
}) => {
  const [isImageUrlValid, setIsImageUrlValid] = useState(true);
  const [showCarousel, setShowCarousel] = useState<boolean>(false);

  const hasImages = imageList && imageList.length > 0;

  useEffect(() => {
    const imageExists = (
      url: string,
      callback: (isImageExists: boolean) => void
    ) => {
      const img: HTMLImageElement = new Image();
      img.onload = () => {
        callback(true);
      };
      img.onerror = () => {
        callback(false);
      };
      img.src = url;
    };

    if (image) {
      imageExists(image, (isImageExists) => {
        setIsImageUrlValid(isImageExists);
      });
    } else {
      setIsImageUrlValid(false);
    }
  }, [image]);

  return (
    <>
      <VehicleImageStyles
        onClick={() =>
          isImageUrlValid && hasImages ? setShowCarousel(true) : {}
        }
      >
        {image && isImageUrlValid ? (
          <>
            <img src={image} className="vehicle-img" alt="" />
            {isImageUrlValid && showImageCount && (
              <div className="image-count">
                <CameraAltIcon className="icon" />
                {imagesCount ?? 0}
              </div>
            )}
          </>
        ) : (
          <div className="image-missing">
            <CarNotFoundIcon />
          </div>
        )}
      </VehicleImageStyles>
      {showCarousel && hasImages && (
        <ImageCarousel
          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
          images={imageList!}
          showCarousel={showCarousel}
          handleClose={() => setShowCarousel(false)}
        />
      )}
    </>
  );
};
