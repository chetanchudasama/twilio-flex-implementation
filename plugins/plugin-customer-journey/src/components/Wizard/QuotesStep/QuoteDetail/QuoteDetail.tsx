import React, { useState } from "react";
import Grid from "@material-ui/core/Grid";
import { Shared } from "@common/components";
import { Button } from "@material-ui/core";
import EditIcon from "@material-ui/icons/Edit";
import CheckCircleIcon from "@material-ui/icons/CheckCircle";
import * as Flex from "@twilio/flex-ui";
import { QuoteDetailModel } from "../../../../models/QuoteDetailModel";
import { VehicleImage } from "../../../VehicleImage/VehicleImage";
import CarDetailViewDialog from "../../../CarDetailViewDialog/CarDetailViewDialog.Container";
import CarDealerInformation from "../../../CarDealerInformation/CarDealerInformation.Container";
import { QuoteDetailStyles } from "./QuoteDetail.Styles";
import QuoteDialog from "../QuoteDialog/QuoteDialog.Container";

export interface QuoteDetailProp {
  quoteDetail: QuoteDetailModel;
}

const QuoteDetail: React.FC<QuoteDetailProp> = ({ quoteDetail }) => {
  const [openCarDetailDialog, setOpenCarDetailDialog] =
    useState<boolean>(false);

  const [openCreateQuoteDialog, setOpenCreateQuoteDialog] =
    useState<boolean>(false);

  const viewDealSheet = () => {
    Flex.Actions.invokeAction("MoveToDealSheetStep", {});
  };

  return (
    <>
      <QuoteDetailStyles>
        <Grid
          container
          className={
            quoteDetail.isAccepted ? "secondary-border" : "grey-border"
          }
        >
          <Grid
            xs={12}
            className={
              quoteDetail.isAccepted
                ? "secondary-background"
                : "grey-background"
            }
            item
          >
            <span className="quote-saved-date">
              {`Saved ${Shared.getFormattedDate(
                quoteDetail.created,
                "DD/MM/YYYY HH:mm"
              )}`}
            </span>
            {quoteDetail.isAccepted && quoteDetail.acceptedDateTime && (
              <span className="customer-accepted-date">
                {`Accepted By Customer ${Shared.getFormattedDate(
                  quoteDetail.acceptedDateTime,
                  "DD/MM/YYYY HH:mm"
                )}`}
              </span>
            )}
            <span className="quote-accepted-msg">
              {quoteDetail.isAccepted && (
                <CheckCircleIcon className="check-icon" />
              )}
              {quoteDetail.isAccepted ? " Approved by 247 Money" : " 247 Money"}
            </span>
          </Grid>
          <Grid item xs={5} className="car-detail-container">
            <Grid container>
              <Grid item xs={12} className="vehicle-img-container">
                <VehicleImage
                  image={quoteDetail.images[0]}
                  imageList={quoteDetail.images}
                  showImageCount={false}
                />
              </Grid>
              <Grid item xs={12} className="car-model">
                {quoteDetail.make}
              </Grid>
              <Grid item xs={12}>
                <ul className="car-specification car-detail">
                  <li>{quoteDetail.model}</li>
                  <li>{quoteDetail.colour}</li>
                  <li>{quoteDetail.doorsCount}dr</li>
                </ul>
              </Grid>
              <Grid item xs={12}>
                <ul className="car-specification">
                  <li>
                    {`${Shared.getFormattedNumber(quoteDetail.mileage)} miles`}
                  </li>
                  <li>{quoteDetail.regNumber}</li>
                  <li>{quoteDetail.regYear}</li>
                  <li>{quoteDetail.trans}</li>
                  <li>{quoteDetail.fuel}</li>
                </ul>
              </Grid>
              <Grid item xs={12} className="view-more-detail">
                <Button
                  className="view-more-detail-link"
                  onClick={() => setOpenCarDetailDialog(true)}
                  disableRipple
                >
                  View More Detail
                </Button>
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={3} className="spec-container">
            <Grid container>
              <Grid item xs={12} className="spec-title">
                Car Value
              </Grid>
              <Grid item xs={12} className="spec-value">
                {Shared.getFormattedCurrencyValue(quoteDetail.price)}
              </Grid>
              <Grid item xs={12} className="spec-title">
                Deposit
              </Grid>
              <Grid item xs={12} className="spec-value">
                {Shared.getFormattedCurrencyValue(quoteDetail.deposit)}
              </Grid>
              <Grid item xs={12} className="spec-title">
                Part Exchange
              </Grid>
              <Grid item xs={12} className="spec-value">
                {Shared.getFormattedCurrencyValue(
                  quoteDetail.partExchangeValue!
                )}
              </Grid>
              <Grid item xs={12} className="spec-title">
                Dealer Fee
              </Grid>
              <Grid item xs={12} className="spec-value">
                -
              </Grid>
              <Grid item xs={12} className="spec-title">
                Paint Protection
              </Grid>
              <Grid item xs={12} className="spec-value">
                {Shared.getFormattedCurrencyValue(
                  quoteDetail.vehicleExtras?.price || 0
                )}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={4} className="finance-section">
            <Grid container className="finance-container">
              <Grid item xs={12} className="offer-term">
                {quoteDetail.term} months
              </Grid>
              <Grid item xs={12} className="offer-price">
                <div className="offer-amount">
                  {Shared.getFormattedCurrencyValue(quoteDetail.priceMonthly)}
                </div>
                <div className="offer-pm"> /pm</div>
              </Grid>
              <Grid item xs={12} className="apr-percentage">
                {quoteDetail.apr}% APR
              </Grid>
              <Grid item xs={12} className="finance-balance">
                {`Balance to finance ${Shared.getFormattedCurrencyValue(
                  quoteDetail.amountToFinance
                )}`}
              </Grid>
              <Grid item xs={12} className="payable-amount">
                {`Total payable ${Shared.getFormattedCurrencyValue(
                  quoteDetail.totalAmountPayable
                )}`}
              </Grid>
              {quoteDetail.isAccepted && (
                <Grid xs={12} item>
                  <Button
                    variant="contained"
                    onClick={viewDealSheet}
                    className="view-deal-sheet-btn"
                  >
                    View deal sheet
                  </Button>
                </Grid>
              )}
              <Grid item xs={12}>
                <Button
                  variant="outlined"
                  className="edit-quote-btn"
                  onClick={() => setOpenCreateQuoteDialog(true)}
                >
                  <EditIcon className="edit-icon" /> Edit quote
                </Button>
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={12} className="car-dealer-info">
            <Grid container>
              <Grid xs={8} item>
                <CarDealerInformation
                  dealerInfo={quoteDetail.dealer}
                  distance={quoteDetail.distance!}
                />
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </QuoteDetailStyles>
      {openCarDetailDialog && (
        <CarDetailViewDialog
          open={openCarDetailDialog}
          vehicleId={quoteDetail.vehicleId}
          handleCloseDialog={() => setOpenCarDetailDialog(false)}
        />
      )}
      {openCreateQuoteDialog && (
        <QuoteDialog
          open={openCreateQuoteDialog}
          handleCloseDialog={() => setOpenCreateQuoteDialog(false)}
          quoteDetail={quoteDetail}
        />
      )}
    </>
  );
};

export default QuoteDetail;
