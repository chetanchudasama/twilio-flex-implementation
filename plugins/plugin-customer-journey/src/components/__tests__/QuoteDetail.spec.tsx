import React from "react";
import { Actions } from "@twilio/flex-ui";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import { QuoteDetailModel } from "../../models/QuoteDetailModel";
import QuoteDetail, {
  QuoteDetailProp,
} from "../Wizard/QuotesStep/QuoteDetail/QuoteDetail";
import { VehicleImage } from "../VehicleImage/VehicleImage";
import CarDealerInformation from "../CarDealerInformation/CarDealerInformation.Container";

jest.mock("@twilio/flex-ui", () => {
  return {
    Actions: { invokeAction: jest.fn(() => {}) },
  };
});

jest.mock("../CarDetailViewDialog/CarDetailViewDialog.Container", () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="car-detail-dialog">Car Detail View Dialog</div>;
    },
  };
});

jest.mock("../CarDealerInformation/CarDealerInformation.Container");
jest.mock("../CarDealerInformation/CarDealerInformation");

jest.mock("../Wizard/QuotesStep/QuoteDialog/QuoteDialog.Container", () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="quote-dialog">Quote Dialog</div>;
    },
  };
});

const props: QuoteDetailProp = {
  quoteDetail: Object.assign(new QuoteDetailModel(), {
    amountToFinance: 10000,
    applicationId: 1,
    apr: 19,
    body: "Estote",
    colour: "Beige",
    created: new Date("2021-07-01 05:30"),
    createdByUserId: 18,
    deposit: 10000,
    distance: 55,
    doorsCount: 5,
    engine: 2,
    fuel: "Diesel Hybrid",
    hasImages: true,
    is247Cars: true,
    isAccepted: true,
    keywords: "mechanic",
    lenderId: 13,
    make: "Land Rover",
    mileage: 13050,
    model: "Discovery Sport",
    monthlyPayment: 4500,
    options: "Dealer-Installed Accessories",
    partExchangeMileage: 2300,
    partExchangeRegistration: "ERT0186980",
    partExchangeSettlement: 1600,
    partExchangeValue: 1600,
    price: 110000,
    priceMonthly: 900,
    quoteId: 5,
    reducedPrice: 90000,
    regNumber: "ST17XYZ",
    regYear: 2011,
    specs: null,
    term: 60,
    tierId: 18,
    totalAmountPayable: 75000,
    trans: "Manual",
    trim: "LX",
    vehicleExtras: {
      vehicleExtraOptionId: 1,
      vehicleExtraTypeId: 1,
      vehicleExtraTypeName: "leather",
      description: "leather",
      price: 100,
    },
    vehicleId: 991,
    year: 2017,
    images: ["Image"],
    dealer: {
      county: "Cheshire",
      dealerId: 16,
      dealerName: "Abby R Farrell",
      phone: "+448580870088",
      postcode: "CW9 4HB",
      rating: "4.5",
      streetName: "41 Lammas Street",
      town: "GREAT BUDWORTH",
      url: "https://namesbee.com/car-dealership-names/",
    },
  }),
};

describe("QuoteDetail", () => {
  it("renders correctly", async () => {
    const wrapper = create(<QuoteDetail {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render VehicleImage component", async () => {
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(VehicleImage).exists()).toBeTruthy();
  });

  it("should render CarDealerInformation component", async () => {
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(CarDealerInformation).exists()).toBeTruthy();
  });

  it("when view more detail link is clicked, CarDetailViewDialog component should be rendered", async () => {
    const wrapper = mount(<QuoteDetail {...props} />);

    wrapper.find(".view-more-detail-link").at(1).simulate("click");

    expect(wrapper.find(".car-detail-dialog").exists()).toBeTruthy();
  });

  it("when value of isAccepted flag is true, proceed with quote button should be displayed", async () => {
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(".view-deal-sheet-btn").exists()).toBeTruthy();
  });

  it("when value of isAccepted flag is false, proceed with quote button should not be displayed", async () => {
    props.quoteDetail.isAccepted = false;
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(".view-deal-sheet-btn").exists()).toBeFalsy();
  });

  it("when proceed with quote button is called, MoveToDealSheetStep Action should be called", async () => {
    props.quoteDetail.isAccepted = true;
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(".view-deal-sheet-btn").exists()).toBeTruthy();

    const spy = jest.spyOn(Actions, "invokeAction");

    wrapper.find(".view-deal-sheet-btn").at(1).simulate("click");

    expect(spy).toHaveBeenCalledWith("MoveToDealSheetStep", {});
  });

  it("when value of isAccepted flag is true, UI should be displayed in purple color", async () => {
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(".secondary-background").exists()).toBeTruthy();
    expect(wrapper.find(".secondary-border").exists()).toBeTruthy();
    expect(wrapper.find(".check-icon").exists()).toBeTruthy();
  });

  it("when value of isAccepted flag is false, UI should be displayed in grey color", async () => {
    props.quoteDetail.isAccepted = false;
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(".grey-background").exists()).toBeTruthy();
    expect(wrapper.find(".grey-border").exists()).toBeTruthy();
    expect(wrapper.find(".check-icon").exists()).toBeFalsy();
  });

  it("when edit quote button is called, Quote Dialog should be displayed", async () => {
    props.quoteDetail.isAccepted = true;
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(".edit-quote-btn").exists()).toBeTruthy();

    wrapper.find(".edit-quote-btn").at(1).simulate("click");

    expect(wrapper.find(".quote-dialog").exists()).toBeTruthy();
  });

  it("when value of acceptedDateTime prop is not null, should render customer accepted date time in UI", async () => {
    props.quoteDetail.isAccepted = true;
    props.quoteDetail.acceptedDateTime = new Date("2021-07-09 05:30");
    const wrapper = mount(<QuoteDetail {...props} />);

    expect(wrapper.find(".customer-accepted-date").exists()).toBeTruthy();

    expect(wrapper.find(".customer-accepted-date").html()).toContain(
      "Accepted By Customer 09/07/2021 05:30"
    );
  });
});
