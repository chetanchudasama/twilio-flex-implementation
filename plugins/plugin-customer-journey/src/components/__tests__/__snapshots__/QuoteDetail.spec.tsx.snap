// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteDetail renders correctly 1`] = `
<div
  className="css-dqpszj"
>
  <div
    className="MuiGrid-container-1 secondary-border"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 secondary-background"
    >
      <span
        className="quote-saved-date"
      >
        Saved 01/07/2021 05:30
      </span>
      <span
        className="quote-accepted-msg"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-98 check-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
         Approved by 247 Money
      </span>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-5-34 car-detail-container"
    >
      <div
        className="MuiGrid-container-1"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 vehicle-img-container"
        >
          <div
            className="css-1epo2pu"
            onClick={[Function]}
          >
            <img
              alt=""
              className="vehicle-img"
              src="Image"
            />
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 car-model"
        >
          Land Rover
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <ul
            className="car-specification car-detail"
          >
            <li>
              Discovery Sport
            </li>
            <li>
              Beige
            </li>
            <li>
              5
              dr
            </li>
          </ul>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <ul
            className="car-specification"
          >
            <li>
              13,050 miles
            </li>
            <li>
              ST17XYZ
            </li>
            <li>
              2011
            </li>
            <li>
              Manual
            </li>
            <li>
              Diesel Hybrid
            </li>
          </ul>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 view-more-detail"
        >
          <button
            className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-text-109 MuiButton-flat-112 view-more-detail-link"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-108"
            >
              View More Detail
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-3-32 spec-container"
    >
      <div
        className="MuiGrid-container-1"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-title"
        >
          Car Value
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-value"
        >
          £110,000
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-title"
        >
          Deposit
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-value"
        >
          £10,000
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-title"
        >
          Part Exchange
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-value"
        >
          £1,600
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-title"
        >
          Dealer Fee
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-value"
        >
          -
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-title"
        >
          Paint Protection
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 spec-value"
        >
          £100
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-4-33 finance-section"
    >
      <div
        className="MuiGrid-container-1 finance-container"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 offer-term"
        >
          60
           months
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 offer-price"
        >
          <div
            className="offer-amount"
          >
            £900
          </div>
          <div
            className="offer-pm"
          >
             /pm
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 apr-percentage"
        >
          19
          % APR
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 finance-balance"
        >
          Balance to finance £10,000
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 payable-amount"
        >
          Total payable £75,000
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <button
            className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-contained-118 MuiButton-raised-121 view-deal-sheet-btn"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-108"
            >
              View deal sheet
            </span>
            <span
              className="MuiTouchRipple-root-136"
            />
          </button>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          <button
            className="MuiButtonBase-root-133 MuiButton-root-107 MuiButton-outlined-115 edit-quote-btn"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-108"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-98 edit-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
               Edit quote
            </span>
            <span
              className="MuiTouchRipple-root-136"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 car-dealer-info"
    >
      <div
        className="MuiGrid-container-1"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
